packages/plugin-electron-deliver-minidumps/minidump-queue.js

```diff
@@ -1,31 +1,21 @@
 module.exports = class MinidumpQueue {
   constructor (filestore) {
     this._filestore = filestore
-    this._minidumps = null
   }
 
   async peek () {
-    if (!this._minidumps) {
-      try {
-        this._minidumps = await this._filestore.listMinidumps()
-      } catch (e) {
-        this._minidumps = []
-      }
+    try {
+      const minidumps = await this._filestore.listMinidumps()
+      return minidumps[0]
```

Comment:
Rather than removing the upload tracking list altogether, it might be worthwhile to track file paths already attempted, in case a file appears at the top of the queue and fails to be deleted, in which case it will block the queue indefinitely.